src/main/java/com/powsybl/openloadflow/network/impl/LfGeneratorImpl.java

```diff
@@ -130,6 +137,13 @@ public double getParticipationFactor() {
         return participationFactor;
     }
 
+    @Override
+    public String getControlledBusId(boolean breakers) {
+        Terminal terminal = generator.getRegulatingTerminal();
+        Bus controlled = breakers ? terminal.getBusBreakerView().getBus() : terminal.getBusView().getBus();
```

Comment:
You have to fix this: you have to check if the bus is not `null` and not creating the voltage control if it is. We do that for the discrete voltage control.

Reply:
Done. In the last commit I do it earlier, in the LfGenerator constructor, so that the voltage control is disabled from the start.